src/LeanCode.Contracts/IProduceNotification.cs

```diff
@@ -2,4 +2,4 @@ namespace LeanCode.Contracts;
 
 [System.Diagnostics.CodeAnalysis.SuppressMessage("?", "CA1040", Justification = "Marker interface.")]
 public interface IProduceNotification<out TNotification>
-    where TNotification : notnull { }
+    where TNotification : class { }
```

Comment:
I am not sure if we still want to block KnownTypes. On our last meeting we decided we are going with the tagging approach (see PIPE-53 on linear and the attached confluence document). Plus, blocking KnownTypes is probably more work than just changing the bound to `class`. `DateOnly`, `List<>`, `Binary` etc are also KnownTypes, but are classes too.

Reply:
Yeah, we agreed upon not blocking the KnownTypes, but supporting tagging `IProduceNotification<T>` types during contract generation so that clients have them listed apriori. This is what we've started talking about today, hit me Wojtek when you need the follow up.

For this PR let's just remove this change.